---
 src/nm-udev-manager.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

Index: b/src/nm-udev-manager.c
===================================================================
--- a/src/nm-udev-manager.c
+++ b/src/nm-udev-manager.c
@@ -284,14 +284,17 @@ static void
 add_one_killswitch (NMUdevManager *self, GUdevDevice *device)
 {
 	NMUdevManagerPrivate *priv = NM_UDEV_MANAGER_GET_PRIVATE (self);
-	const char *str_type;
+	const char *str_type, *driver;
 	RfKillType rtype;
 	Killswitch *ks;
 
 	str_type = g_udev_device_get_property (device, "RFKILL_TYPE");
+	driver = g_udev_device_get_property (device, "DRIVER");
 	rtype = rfkill_type_to_enum (str_type);
 	if (rtype == RFKILL_TYPE_UNKNOWN)
 		return;
+	if (!strcmp (driver, "wcnss_wlan"))
+		return;
 
 	ks = killswitch_new (device, rtype);
 	priv->killswitches = g_slist_prepend (priv->killswitches, ks);
