From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: Ignore rmnet_usbX devices.
---
 src/nm-device-ethernet.c |    4 ++++
 1 file changed, 4 insertions(+)

Index: b/src/nm-device-ethernet.c
===================================================================
--- a/src/nm-device-ethernet.c
+++ b/src/nm-device-ethernet.c
@@ -502,6 +502,10 @@ get_generic_capabilities (NMDevice *dev)
 	if (strstr (nm_device_get_iface (dev), "cipsec"))
 		return NM_DEVICE_CAP_NONE;
 
+	/* Don't support rmnet_usb; just let them available for ril to play with */
+	if (strstr (nm_device_get_iface (dev), "rmnet_usb"))
+		return NM_DEVICE_CAP_NONE;
+
 	if (supports_ethtool_carrier_detect (self) || supports_mii_carrier_detect (self))
 		caps |= NM_DEVICE_CAP_CARRIER_DETECT;
 
