Index: nm-0.9.6.0+git201301221753.{revno}-0~pkg{revno:debian}/src/nm-manager.c
===================================================================
--- nm-0.9.6.0+git201301221753.{revno}-0~pkg{revno:debian}.orig/src/nm-manager.c	2013-01-22 12:55:32.353177930 -0500
+++ nm-0.9.6.0+git201301221753.{revno}-0~pkg{revno:debian}/src/nm-manager.c	2013-01-22 13:26:36.849268166 -0500
@@ -1665,6 +1665,17 @@
 		return;
 	}
 
+	/*
+	 * Explicitly ignore p2p Wi-Fi devices exposed by Android JB Wi-Fi drivers.
+	 */
+	if ((devtype == NM_DEVICE_TYPE_WIFI) && g_strcmp0 ("p2p0", iface) == 0) {
+		nm_log_info (LOGD_HW, "(%s): ignoring P2P wireless iface (ifindex: %d)",
+					 iface, nm_device_get_ifindex (device));
+
+		g_object_unref (device);
+		return;
+	}
+
 	nm_device_set_connection_provider (device, NM_CONNECTION_PROVIDER (priv->settings));
 
 	priv->devices = g_slist_append (priv->devices, device);
