Index: network-manager-0.9.0/src/nm-device.c
===================================================================
--- network-manager-0.9.0.orig/src/nm-device.c	2011-08-30 12:54:16.827044560 -0400
+++ network-manager-0.9.0/src/nm-device.c	2011-08-30 12:55:25.677044691 -0400
@@ -810,8 +810,8 @@
 		g_assert (method);
 	}
 
-	/* Treat missing IP6 setting as IGNORE */
-	if (!s_ip6 && !strcmp (match, NM_SETTING_IP6_CONFIG_METHOD_IGNORE))
+	/* Treat missing IP6 setting as AUTO */
+	if (!s_ip6 && !strcmp (match, NM_SETTING_IP6_CONFIG_METHOD_AUTO))
 		return TRUE;
 
 	return method && !strcmp (method, match);
