From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: Prioritize Wifi over 3G, if only because it's cheaper.

---
 src/nm-device.c |   16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

Index: b/src/nm-device.c
===================================================================
--- a/src/nm-device.c
+++ b/src/nm-device.c
@@ -688,6 +688,10 @@ nm_device_get_priority (NMDevice *dev)
 	 *     it's using that connection.
 	 */
 
+	/* Ubuntu change:
+	 * Adjust priorities to de-prioritize bluetooth, wimax and modem
+	 * to lower; BT is just slow, WiMAX and "MODEM" because they cost
+	 * money. */
 	switch (nm_device_get_device_type (dev)) {
 	case NM_DEVICE_TYPE_ETHERNET:
 		return 1;
@@ -695,20 +699,20 @@ nm_device_get_priority (NMDevice *dev)
 		return 2;
 	case NM_DEVICE_TYPE_ADSL:
 		return 3;
-	case NM_DEVICE_TYPE_WIMAX:
-		return 4;
 	case NM_DEVICE_TYPE_BOND:
 		return 5;
 	case NM_DEVICE_TYPE_VLAN:
 		return 6;
-	case NM_DEVICE_TYPE_MODEM:
-		return 7;
-	case NM_DEVICE_TYPE_BT:
-		return 8;
 	case NM_DEVICE_TYPE_WIFI:
 		return 9;
 	case NM_DEVICE_TYPE_OLPC_MESH:
 		return 10;
+	case NM_DEVICE_TYPE_BT:
+		return 11;
+	case NM_DEVICE_TYPE_WIMAX:
+		return 12;
+	case NM_DEVICE_TYPE_MODEM:
+		return 13;
 	default:
 		return 20;
 	}
